### ISSUE TITLE
sphinx viewcode fails intermittently during build

### ISSUE THREAD

**Issue by axnsan12** [rtfd/readthedocs.org#3411]
## Details

* Project URL: https://drf-yasg.readthedocs.io/
* Build URL (if applicable): https://readthedocs.org/projects/drf-yasg/builds/6423001/, https://readthedocs.org/projects/drf-yasg/builds/6423014/
* Read the Docs username (if applicable): cvijdea

## Expected Result

Build completes succesfully and generates HTML of source code.

It builds just fine in my local virtualenv and on Travis:
https://travis-ci.org/axnsan12/drf-yasg/jobs/317361566

## Actual Result

```pytb
python /home/docs/checkouts/readthedocs.org/user_builds/drf-yasg/envs/latest/bin/sphinx-build -T -b readthedocs -d _build/doctrees-readthedocs -D language=en . _build/html
Running Sphinx v1.6.5
making output directory...
loading translations [en]... done
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [readthedocs]: targets for 9 source files that are out of date
updating environment: [config changed] 10 added, 0 changed, 0 removed
reading sources... [ 10%] changelog
reading sources... [ 20%] contributing
reading sources... [ 30%] custom_spec
reading sources... [ 40%] custom_ui
reading sources... [ 50%] drf_yasg

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/drf-yasg/envs/latest/lib/python3.5/site-packages/sphinx/cmdline.py", line 306, in main
    app.build(opts.force_all, filenames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/drf-yasg/envs/latest/lib/python3.5/site-packages/sphinx/application.py", line 339, in build
    self.builder.build_update()
  File "/home/docs/checkouts/readthedocs.org/user_builds/drf-yasg/envs/latest/lib/python3.5/site-packages/sphinx/builders/__init__.py", line 331, in build_update
    'out of date' % len(to_build))
  File "/home/docs/checkouts/readthedocs.org/user_builds/drf-yasg/envs/latest/lib/python3.5/site-packages/sphinx/builders/__init__.py", line 344, in build
    updated_docnames = set(self.env.update(self.config, self.srcdir, self.doctreedir))
  File "/home/docs/checkouts/readthedocs.org/user_builds/drf-yasg/envs/latest/lib/python3.5/site-packages/sphinx/environment/__init__.py", line 584, in update
    self._read_serial(docnames, self.app)
  File "/home/docs/checkouts/readthedocs.org/user_builds/drf-yasg/envs/latest/lib/python3.5/site-packages/sphinx/environment/__init__.py", line 603, in _read_serial
    self.read_doc(docname, app)
  File "/home/docs/checkouts/readthedocs.org/user_builds/drf-yasg/envs/latest/lib/python3.5/site-packages/sphinx/environment/__init__.py", line 735, in read_doc
    app.emit('doctree-read', doctree)
  File "/home/docs/checkouts/readthedocs.org/user_builds/drf-yasg/envs/latest/lib/python3.5/site-packages/sphinx/application.py", line 489, in emit
    return self.events.emit(event, self, *args)
  File "/home/docs/checkouts/readthedocs.org/user_builds/drf-yasg/envs/latest/lib/python3.5/site-packages/sphinx/events.py", line 79, in emit
    results.append(callback(*args))
  File "/home/docs/checkouts/readthedocs.org/user_builds/drf-yasg/envs/latest/lib/python3.5/site-packages/sphinx/ext/viewcode.py", line 100, in doctree_read
    if not has_tag(modname, fullname, env.docname, refname):
  File "/home/docs/checkouts/readthedocs.org/user_builds/drf-yasg/envs/latest/lib/python3.5/site-packages/sphinx/ext/viewcode.py", line 74, in has_tag
    if entry is None or entry[0] != code:
TypeError: 'bool' object is not subscriptable

Exception occurred:
  File "/home/docs/checkouts/readthedocs.org/user_builds/drf-yasg/envs/latest/lib/python3.5/site-packages/sphinx/ext/viewcode.py", line 74, in has_tag
    if entry is None or entry[0] != code:
TypeError: 'bool' object is not subscriptable
The full traceback has been saved in /tmp/sphinx-err-3hfj2uz4.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks! 

```

## **UPDATE** 

After retrying multiple times, I managed to get a succesful build: https://readthedocs.org/projects/drf-yasg/builds/6423027/

I also disabled PDF generation in the meantime and this is (perhaps coincidentally) the first build with it disabled.

## **UPDATE 2**

Build success is still a lottery so I guess disabling PDF generation had no effect.

**Comment by axnsan12** [comment]
Wiping all build environments seems to have fixed this...

**Comment by glyg** [comment]
I got the same error today on a python 3.6 / sphinx 1.6.3 env. Maybe worth adding a short comment to the error message saying 'please wipe your build'? I can open a PR for that if ok.

**Comment by RichardLitt** [comment]
@glyg That makes sense to me! Can you open a PR?

**Comment by glyg** [comment]
Ok, turned out to be a fix, not just some editing of the warning:

https://github.com/sphinx-doc/sphinx/pull/4397

Thanks for the work on those tools!

**Comment by mxbi** [comment]
Had the same error today. Worked on the third attempt just rerunning `make.bat html`, without wiping anything. Sphinx v1.5.6

**Comment by axnsan12** [comment]
@mxbi @glyg's fix was [merged into sphinx 1.6.6](https://github.com/sphinx-doc/sphinx/commit/5e4806a764ded263060afbbde70beced80867a22). You should update your Sphinx.
